### DISCUSSION TITLE
Low FPS on mod select screen on intel IGPU

### DISCUSSION THREAD

**Discussion by Tskk-dev** [ppy/osu#25314] [Q&A] [answered]
### Type

Performance

### Bug description

FPS is fine on song select but as soon as I go to mod select the FPS drops by a lot, Doesn't seem to be much of an issue on multi threaded but is exemplified on the single threaded setting.

### song select
![osu_2023-10-31_10-56-27](https://github.com/ppy/osu/assets/96981837/8dd398e0-45c3-479a-9d5b-7e2f8c9f4528)

### mod select
![osu_2023-10-31_10-56-55](https://github.com/ppy/osu/assets/96981837/efb869cb-fb44-4956-8b7e-82c7a836211b)

### Version

2023.1026.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/13211570/database.log)
[input.log](https://github.com/ppy/osu/files/13211571/input.log)
[performance.log](https://github.com/ppy/osu/files/13211573/performance.log)
[runtime.log](https://github.com/ppy/osu/files/13211574/runtime.log)
[updater.log](https://github.com/ppy/osu/files/13211575/updater.log)

**Comment by peppy** [comment]
What refresh rate is your screen? It looks to still be running at 60fps, which should be enough on a 60 hz monitor.

**Reply by Tskk-dev** [reply]
60hz, the actual game feels sluggish while on the mod select screen tho, gets worse when I scroll horizontally, here is a recording, should still show the effect even if im taking a bit of a hit from recording, can close as resolved if expected behavior tho.

https://github.com/ppy/osu/assets/96981837/b92b2593-90ea-4172-8e7c-b8709e11637b

**Reply by peppy** [reply, marked as answer]
Looks like the game is hitting fill rate / shader limits. Given it's a 2017 GPU, this is definitely underneath our performance benchmark, so I hope we can improve things in the future.

**Comment by Tskk-dev** [comment]
I came back to this after 2 years to say that this issue has been completely fixed! 
<img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/2b37519f-a974-47d5-998e-fa1005711c8e" />
